05/app.js

```diff
@@ -0,0 +1,63 @@
+const formEl = document.querySelector('form');
+const messagesList = document.querySelector('.messages');
+formEl.noValidate = true;
+
+if (formEl) {
+    formEl.addEventListener('submit', checkData)
+}
+
+function createMessageItem(message) {
+    const listItem = document.createElement('li');
+
+    listItem.classList.add('messages__item');
+    listItem.innerText = message;
+
+    return listItem;
+}
+
+function checkData(e) {
+    e.preventDefault();
+
+    const messages = [];
+
+    for (const key of formEl) {
+        const label = key.previousSibling;
+        const labelParent = key.parentElement;
+
+        if (key.hasAttribute('required') && key.value.length === 0) {
+            labelParent.style.color = 'red';
+            messages.push(`Pole: ${label.textContent} jest wymagane!`);
+        } else {
+            labelParent.style.color = 'black';
+        }
+        if (key.type === 'number' && key.value.length > 0 && key.value <= 0) {
+            labelParent.style.color = 'red';
+            messages.push(`Podaj poprawny ${label.textContent}!`);
+        }
+        if (key.name === 'zip') {
+            const value = key.value;
+            const isValidZip = /^[0-9]{2}(?:-[0-9]{3})?$/.test(value);
+
+            if (!isValidZip) {
+                labelParent.style.color = 'red';
+                messages.push(`Podaj poprawny ${label.textContent}!`)
+            }
+        }
```

Comment:
ok, to może teraz wykorzystasz jeszcze bardziej uniwersalne rozwiązanie, o którym przeczytasz tutaj:
https://devmentor.pl/b/walidacja-formularza-w-javascript